api/docs/mapstruct.md

```diff
@@ -0,0 +1,608 @@
+
+    private final SampleEnumMapper sampleEnumMapper = Mappers.getMapper( SampleEnumMapper.class );
+
+    @Override
+    public EmployeeModel toModel(EmployeeDbEntity employeeDbEntity) {
+        if ( employeeDbEntity == null ) {
+            return null;
+        }
+
+        EmployeeModel employeeModel = new EmployeeModel();
+
+        employeeModel.setName( toNickname( employeeDbEntity ) );
+        employeeModel.setDepartment( sampleEnumMapper.toDepartment( employeeDbEntity.getDepartmentCode() ) );
+        employeeModel.setSalary( toAnnalSalary( employeeDbEntity.getHourlyRate() ) );
+        employeeModel.setAddress( employeeDbEntity.getAddress() );
+
+        return employeeModel;
+    }
+}
+```
+
+### Testing
+It's pretty easy to add a test class for the mapper. Here's a basic test case for the happy path.
+If your input class has any nullable parameters or fields on those parameters, or there are any other
+edge cases, you'd want to test those as well.
+```java
+import static com.google.common.truth.Truth.assertThat;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.pmiops.workbench.utils.mappers.examples.Department;
+import org.pmiops.workbench.utils.mappers.examples.EmployeeDbEntity;
+import org.pmiops.workbench.utils.mappers.examples.EmployeeMapper;
+import org.pmiops.workbench.utils.mappers.examples.EmployeeMapperImpl;
+import org.pmiops.workbench.utils.mappers.examples.EmployeeModel;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.TestConfiguration;
+import org.springframework.context.annotation.Import;
+import org.springframework.test.context.junit4.SpringRunner;
+
+@RunWith(SpringRunner.class)
+public class EmployeeMapperTest {
+
+  public static final String ADDRESS = "123 Sesame Street";
+  @Autowired private EmployeeMapper employeeMapper;
+
+  @TestConfiguration
+  @Import({EmployeeMapperImpl.class})
+  static class Configuration {}
+
+  @Test
+  public void testToModel() {
+    final EmployeeDbEntity employeeDbEntity = new EmployeeDbEntity("John", "Doe", 2, 3.00, ADDRESS);
+
+    final EmployeeModel employeeModel = employeeMapper.toModel(employeeDbEntity);
+    assertThat(employeeModel.getDepartment()).isEqualTo(Department.IT);
+    assertThat(employeeModel.getSalary()).isWithin(1.0e-6).of(6000.0);
+    assertThat(employeeModel.getName()).isEqualTo("John Doe");
+    assertThat(employeeModel.getAddress()).isEqualTo(ADDRESS);
+  }
+}
+```
+## Tips & Strategies
+
+### Stub it so it builds, and go one property at a time
+For big target classes, put in some temporary `ignore` statements or other hacks just to get it
+to the point that it generates the implemmentation class and actually compiles without errors. Then
+proceed one property at a time until all the warnings are gone.
+ 
+### Proofread the generated code
+First, you should *always* look at the generated code and step through it in the debugger. This is
```

Comment:
It would be great to mention some sample file names and where they appear in the filetree here. (Filetree optional, since it could move and make our documentation stale)

Reply:
I yanked them from the repo. I'd need to make a new project and put them there. In the end I decided this article could be the source of truth.

If you just make a new file in the existing mappers directory (or anywhere) it'll work. Then put the test case in the same package under the test tree.